pkg/service/v0/service.go

```diff
@@ -23,30 +27,84 @@ func NewService(opts ...Option) Service {
 	svc := Thumbnails{
 		config: options.Config,
 		mux:    m,
+		manager: thumbnails.SimpleManager{
+			Storage: storage.NewFileSystemStorage(options.Config.FileSystemStorage),
+		},
+		source: imgsource.NewWebDavSource(options.Config.WebDavSource),
 	}
 
 	m.Route(options.Config.HTTP.Root, func(r chi.Router) {
-		r.Get("/", svc.Dummy)
+		r.Get("/thumbnails", svc.Thumbnails)
 	})
 
 	return svc
 }
 
 // Thumbnails defines implements the business logic for Service.
 type Thumbnails struct {
-	config *config.Config
-	mux    *chi.Mux
+	config  *config.Config
+	mux     *chi.Mux
+	manager thumbnails.Manager
+	source  imgsource.Source
 }
 
 // ServeHTTP implements the Service interface.
 func (g Thumbnails) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	g.mux.ServeHTTP(w, r)
 }
 
-// Dummy implements the Service interface.
-func (g Thumbnails) Dummy(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "text/plain")
-	w.WriteHeader(http.StatusOK)
+// Thumbnails provides the endpoint to retrieve a thumbnail for an image
+func (g Thumbnails) Thumbnails(w http.ResponseWriter, r *http.Request) {
+	query := r.URL.Query()
+	width, _ := strconv.Atoi(query.Get("width"))
+	height, _ := strconv.Atoi(query.Get("height"))
+	fileType := query.Get("type")
+	filePath := query.Get("file_path")
+	etag := query.Get("etag")
```

Comment:
Generally speaking I'd declare a `struct` and try to unmarshal the request query string onto it, then operate with typed fields. It is the same concept you're doing here but a bit more explicit. We could give it a try together